presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -794,6 +800,10 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
             return InstanceProfileCredentialsProvider.getInstance();
         }
 
+        if (s3IamRole != null) {
+            return new STSAssumeRoleSessionCredentialsProvider.Builder(this.s3IamRole, "presto-session").build();
```

Comment:
same for this.s3IamRole
shall we have a `roleSessionName `, instead of using `presto-session`?

Reply:
following https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html
do you think it will be valuable to specify maybe something like "presto - " + uri as the roleSessionName here instead of presto-session to also have the accessed uri as part of the role name?

Reply:
Agree.. would be good for audit. changed to `"presto-" + uri`